contracts/orderbook/Logic/OrderBookData.sol

```diff
@@ -0,0 +1,117 @@
+        _setTarget(this.getActiveOrdersLimited.selector, target);
+        _setTarget(this.getOrderByOrderID.selector, target);
+        _setTarget(this.getActiveOrderIDs.selector, target);
+        _setTarget(this.getTotalOrders.selector, target);
+        _setTarget(this.getActiveOrders.selector, target);
+        _setTarget(this.getTotalActiveOrders.selector, target);
+        _setTarget(this.getOrders.selector, target);
+        _setTarget(this.getOrdersLimited.selector, target);
+    }
+
+    function adjustAllowance(address spender, address token) external pausable {
+        require(
+            protocol.isLoanPool(spender) ||
+                address(protocol) == spender ||
+                vault == spender,
+            "OrderBook: invalid spender"
+        );
+        IERC(token).approve(spender, 0); //needs to be zeroed out because of different iterations of ERC-20 standard
+        IERC(token).approve(spender, type(uint256).max);
+    }
+
+    function getActiveOrders(address trader)
+        external
+        view
+        returns (IOrderBook.Order[] memory fullList)
+    {
+        bytes32[] memory idSet = _histOrders[trader].values();
+
+        fullList = new IOrderBook.Order[](idSet.length);
+        for (uint256 i = 0; i < idSet.length;) {
+            fullList[i] = _allOrders[idSet[i]];
+            unchecked { ++i; }
+        }
+        return fullList;
+    }
+
+    function getActiveOrdersLimited(address trader, uint start, uint end)
+        external
+        view
+        returns (IOrderBook.Order[] memory fullList)
+    {
+        require(end<=_histOrders[trader].length(), "OrderBook: end is past max orders");
+        fullList = new IOrderBook.Order[](end-start);
+        for (uint256 i = start; i < end;) {
+            fullList[i] = _allOrders[_histOrders[trader].at(i)];
+            unchecked { ++i; }
+        }
+        return fullList;
+    }
+
+    function getOrderByOrderID(bytes32 orderID)
+        public
+        view
+        returns (IOrderBook.Order memory)
+    {
+        return _allOrders[orderID];
+    }
+
+    function getActiveOrderIDs(address trader)
+        external
+        view
+        returns (bytes32[] memory)
+    {
+        return _histOrders[trader].values();
+    }
+
+    function getTotalOrders(address trader) external view returns (uint256) {
+        return _histOrders[trader].length();
+    }
+
+    function getTotalActiveOrders() external view returns (uint256) {
+        return _allOrderIDs.length();
+    }
+
+    function getOrders()
+        external
+        view
+        returns (IOrderBook.Order[] memory fullList)
+    {
+        bytes32[] memory idSet = _allOrderIDs.values();
+
+        fullList = new IOrderBook.Order[](idSet.length);
+        for (uint256 i = 0; i < idSet.length;) {
+            fullList[i] = getOrderByOrderID(idSet[i]);
+            unchecked { ++i; }
+        }
+        return fullList;
+    }
+
+    function getOrdersLimited(uint start, uint end)
```

Comment:
maybe we can have one more func `getOrdersLimitedIDs` would be handy in upkeep since you won't have to pass lots of data in a return

Reply:
probably return `_allOrderIDs`

Reply:
I think it is nice to have but not entirely necessary. Will add anyways